roles/variables/vars/main.yml

```diff
@@ -259,6 +259,10 @@ kafka_controller_properties:
     enabled: "{{oauth_enabled and oauth_groups_claim != 'none'}}"
     properties:
       confluent.oauth.groups.claim.name: "{{oauth_groups_claim}}"
+  listener_oauth:
+    enabled: "{{ oauth_enabled|bool or sso_mode != 'none' }}"
+    properties: # will need to be updated if using controller listener of some other name
+      listener.name.controller.principal.builder.class: io.confluent.kafka.security.authenticator.OAuthKafkaPrincipalBuilder
```

Comment:
this OAuthKafkaPrincipalBuilder is getting added from filters.py for brokers. Should this also be added from that file to keep it similar in structure or there is some specific reason to add it here

Reply:
Yes, I thought about it but the issue here is if we check the condition that this listener should not have OAUTHBEARER and oauth should be enabled, then it will be satisfied for controller. But at the same time it will be satisfied for other listeners also like a plain listener on broker. There is no way to identify that the listener is a controller listener. You think adding a flag for controller listener would be a better approach?

Reply:
i think checking for oauth_enabled and not oauthbearer will not be a good way. so maybe adding a flag can be done but you take the call you think that will be better for longer run then we can do it or else this is also fine

Reply:
agree, adding a flag would help in the longer run when we make controller listener modifiable. Let me change that